### PR TITLE
Check file_types is a list

### LINKED ISSUE #3043: `gr.File` with `file_types="image"` breaks front-end rendering
### Describe the bug

If a `gr.File` contains the `file_types="image"` the front-end never renders due to a `Uncaught (in promise) TypeError: i.map is not a function` in the browser console
image

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

```py
import gradio as gr
with gr.Blocks() as demo:
    gr.File(label="Test", file_count="multiple", interactive=True, file_types="image") 

demo.launch()
```
https://huggingface.co/spaces/multimodalart/grFileBug

### Screenshot

_No response_

### Logs

```shell
FileUpload.svelte:49 Uncaught (in promise) TypeError: i.map is not a function
    at Ce (FileUpload.svelte:49:27)
    at Pt (index.4395ab38.js:4:6005)
    at new Ee (FileUpload.svelte:60:30)
    at Array.Me (File.svelte:57:16)
    at Array.Se (File.svelte:40:16)
    at jn (index.4395ab38.js:1:1425)
    at _n (index.4395ab38.js:76:13914)
    at Wc (index.4395ab38.js:76:14690)
    at Pt (index.4395ab38.js:4:6206)
    at new Zc (index.4395ab38.js:76:16107)
```
### System Info

```shell
gradio==3.16.2
```

### Severity

blocking upgrade to latest gradio version

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

Fixes #3043 

As noted in #3043, if `file_types` is not a list, the app will crash. Raising an error on the back-end to check that parameter is a list!

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

File: gradio/components.py [hunk 1]
```diff
@@ -2250,6 +2250,10 @@ def __init__(
         """
         self.file_count = file_count
         self.file_types = file_types
+        if file_types is not None and not isinstance(file_types, list):
+            raise ValueError(
+                f"Parameter file_types must be a list. Received {file_types.__class__.__name__}"
```
Comment (abidlabs) [inline gradio/components.py:2255]: Did not know you could do this on primitives, nice!

**Comment by abidlabs** [comment]
Thanks @freddyaboulton, this looks good! Can you also add this logic to `gr.UploadButton()`?

**Comment by freddyaboulton** [comment]
@abidlabs - good point! Just added that logic to UploadButton!

**Review by abidlabs — APPROVED** [review]
Thanks @freddyaboulton LGTM!

**Comment by freddyaboulton** [comment]
Thank you for the review @abidlabs !

**Pull request merged**